src/contractor/contractor.cpp

```diff
@@ -494,6 +541,23 @@ std::size_t Contractor::LoadEdgeExpandedGraph(
                 previous_osm_node_id = this_osm_node_id;
             }
 
+            OSMNodeID from_id;
+            OSMNodeID via_id;
+            OSMNodeID to_id;
+            edge_fixed_penalties_input_stream.read(reinterpret_cast<char *>(&from_id),
+                                                   sizeof(from_id));
+            edge_fixed_penalties_input_stream.read(reinterpret_cast<char *>(&via_id),
+                                                   sizeof(via_id));
+            edge_fixed_penalties_input_stream.read(reinterpret_cast<char *>(&to_id), sizeof(to_id));
+
+            auto turn_iter = turn_penalty_lookup.find(std::make_tuple(from_id, via_id, to_id));
+            if (turn_iter != turn_penalty_lookup.end())
+            {
+                int new_turn_weight = static_cast<int>(turn_iter->second.first * 10);
+                new_weight += new_turn_weight;
+            }
+
+            BOOST_ASSERT(static_cast<int>(fixed_penalty) + new_weight > 0);
             inbuffer.weight = fixed_penalty + new_weight;
```

Comment:
We should use *either* `new_turn_weight`, or `fixed_penalty` here.

The `fixed_penalty` value is the original turn_penalty + stop_sign penalty + u_turn_penalty.  To be absolutely correct, we should split `fixed_penalty` into those three parts, save them, then add the proper ones back here.

However, I suspect very few intersections have all three values applied at the same time, so we're probably save to just use `new_turn_weight` instead of `fixed_penalty` when we find an intersection match.